pkg/reconciler/build/resources/pod.go

```diff
@@ -249,6 +250,15 @@ func makeCredentialInitializer(build *v1alpha1.Build, kubeclient kubernetes.Inte
 func MakePod(build *v1alpha1.Build, kubeclient kubernetes.Interface) (*corev1.Pod, error) {
 	build = build.DeepCopy()
 
+	// Copy annotations on the build through to the underlying pod to allow users
+	// to specify pod annotations.
+	annotations := map[string]string{
+		"sidecar.istio.io/inject": "false",
+	}
+	for key, val := range build.Annotations {
+		annotations[key] = val
```

Comment:
If build has annotation `sidecar.istio.io/inject` set to `true` then previous annotation that sets to false is overridden. Is that desired behavior ?
If we are sure that `sidecar.istio.io/inject` should always be set to `false` then probably set this key after reading build annotations.  WDYT? @mathspanda @ImJasonH

Reply:
Yeap. I think you are right. `sidecar.istio.io/inject` should be set last. Fixed.